crates/prql-compiler/src/parser.rs

```diff
@@ -34,9 +31,11 @@ fn parse_source(source: &str, source_id: u16) -> Result<Vec<prql_ast::stmt::Stmt
         Errors(
             errors
                 .into_iter()
-                .map(|err| match err {
-                    prql_parser::Error::Lexer(err) => convert_lexer_error(source, err, source_id),
-                    prql_parser::Error::Parser(err) => convert_parser_error(err),
+                .map(|err| {
+                    // TODO: we actually want to to avoid this stringification
+                    // here but it's currently just there because Reason
+                    // currently must implement Clone but we don't necessarily want prql_parser::Error to implement Clone.
```

Comment:
Thanks for the code comment. 

Would there be a downside of `prql_parser::Error` implementing `Clone`? 

If it did, how would we avoid the stringification? Wouldn't `reason` still need to be a flat string?

Reply:
I think the only reason why the compiler error type implements `Clone` is because of the downcasting that `WithErrorInfo` does ... we should get rid of this downcasting anyway, at which point all the Clone impls would be rather redundant. But adding it now and removing it later would be a breaking change, so I'd rather not add it at all.

The idea for the future is to rename `Reason` to `ErrorKind` and have a variant like `Parser(prql_parser::Error)`, so the stringification would only happen on `Display`.

Reply:
What's the downside of implementing `Clone` though?

Reply:
Derive impls increase the compilation time ... for this particular instance probably rather negligible but I like to in general avoid derive impls that aren't actually needed.

Reply:
I see, thanks